File: code/front-end/src/app/models/comment.ts
import {Book} from './book';
import {Movie} from './movie';

export class Comment {
  id?: number;
  title: string;
  body: string;
  date: Date;
  book?: Book;
  movie?: Movie;
  constructor(item: any) {
    if (typeof(item) === 'number') {
      this.id = item;
    } else {
      this.id = item.Id;
      this.title = item.Name;
      this.body = item.Description;
      this.date = item.Date;
      this.book = item.Book;
      this.movie = item.Movie;
    }
  }
}

File: code/back-end/TellYourFriends/TellYourFriends/Models/Entity/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace TellYourFriends.Models.Entity
{
    public class Comment
    {
        [Key]
        public int Id { get; set; }

        public string Title { get; set; }

        public string Body { get; set; }

        public DateTime Date { get; set; }

        public int Likes { get; set; }

        public Book Book { get; set; }
        
        public Movie Movie { get; set; }
    }
}